haystack/components/rankers/transformers_similarity.py

```diff
@@ -73,46 +73,61 @@ def __init__(
         """
         torch_and_transformers_import.check()
 
-        self.model = model
-        if top_k <= 0:
-            raise ValueError(f"top_k must be > 0, but got {top_k}")
+        self.model_name_or_path = str(model)
+        self.model = None
+        self.tokenizer = None
         self.top_k = top_k
-        self.device = ComponentDevice.resolve_device(device)
         self.token = token
-        self._model = None
-        self.tokenizer = None
         self.meta_fields_to_embed = meta_fields_to_embed or []
         self.embedding_separator = embedding_separator
         self.scale_score = scale_score
         self.calibration_factor = calibration_factor
-        if self.scale_score and self.calibration_factor is None:
-            raise ValueError(
-                f"scale_score is True so calibration_factor must be provided, but got {calibration_factor}"
-            )
         self.score_threshold = score_threshold
         self.model_kwargs = model_kwargs or {}
 
-        if self.device.has_multiple_devices:
+        if self.model_kwargs.get("device_map") and device is not None:
+            raise ValueError(
+                "The parameters `device` and `device_map` from `model_kwargs` cannot both be provided."
+                "Provide only one or the other."
+            )
```

Comment:
The HF local generator allows the user to override the explicit `device` parameter in `hugging_face_pipeline_args`. I think it would be more consistent to go for that usage pattern? We discussed this internally and sort of landed on the following separation:
- The `ComponentDevice` abstraction allows for a baseline selection of single and multiple devices. Generally, we want to steer users to use this abstraction.
- More advanced users can just pass the framework-specific device/device map as the `model/pipeline_kwargs`. This is also the way of passing device parameters that `ComponentDevice` does not/cannot support.

Reply:
Sounds good! Unfortunately I can't currently use the logic 
```python
device = ComponentDevice.resolve_device(device)
device.update_hf_kwargs(huggingface_pipeline_kwargs, overwrite=False)
```
like in the HF local generator because `update_hf_kwargs` logic only works for pipelines, so these lines 
```python
        converted = self.to_hf()
        key = "device_map" if self.has_multiple_devices else "device"
        hf_kwargs[key] = converted
        return hf_kwargs
```
won't work, because we can't provide `device` as kwarg when using things like `AutoModelForQuestionAnswering.from_pretrained(**model_kwargs)`. It only accepts `device_map` as a kwarg, `device` will throw an error.

Reply:
This isn't an issue for the HF Local generator since it uses the HF pipelines object which handles the first_device logic internally. Unfortunately, for our components that use the AutoModelFor... class we have to handle this `first_device` logic.